app/models/user.rb

```diff
@@ -360,9 +350,22 @@ def self.user_tips
       )
   end
 
+  def secured_sidebar_category_ids(user_guardian = nil)
+    user_guardian ||= guardian
+
+    SidebarSectionLink.where(user_id: self.id, linkable_type: "Category").pluck(:linkable_id) &
```

Comment:
```suggestion
    SidebarSectionLink.where(user: self, linkable_type: "Category").pluck(:linkable_id) &
```

Reply:
O this is different because we want to pluck out the category ids whereas `visible_sidebar_tags` is using a subquery.